src/test/java/edu/ucsb/cs156/example/controllers/RecommendationRequestsControllerTests.java

```diff
@@ -0,0 +1,351 @@
+                LocalDateTime dateNeeded = LocalDateTime.parse("2022-01-05T00:00:00");
+
+
+                RecommendationRequest recommendationRequest1 = RecommendationRequest.builder()
+                                .requesterEmail("supman@gmail.com")
+                                .professorEmail("drbob@ucsb.edu")
+                                .explanation("idk")
+                                .dateRequested(dateRequested)
+                                .dateNeeded(dateNeeded)
+                                .done(false)
+                                .build();
+
+                LocalDateTime dateRequested2 = LocalDateTime.parse("2022-01-03T00:00:00");
+                LocalDateTime dateNeeded2 = LocalDateTime.parse("2022-01-10T00:00:00");
+
+
+                RecommendationRequest recommendationRequest2 = RecommendationRequest.builder()
+                                .requesterEmail("please@yahoo.com")
+                                .professorEmail("drperky@ucsb.edu")
+                                .explanation("please man")
+                                .dateRequested(dateRequested2)
+                                .dateNeeded(dateNeeded2)
+                                .done(true)
+                                .build();
+
+                ArrayList<RecommendationRequest> expectedRecommendationRequests = new ArrayList<>();
+                expectedRecommendationRequests.addAll(Arrays.asList(recommendationRequest1, recommendationRequest2));
+
+                when(recommendationRequestRepository.findAll()).thenReturn(expectedRecommendationRequests);
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/recommendationrequests/all"))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+
+                verify(recommendationRequestRepository, times(1)).findAll();
+                String expectedJson = mapper.writeValueAsString(expectedRecommendationRequests);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void an_admin_user_can_post_a_new_recommendation_request() throws Exception {
+                // arrange
+
+                LocalDateTime dateRequested = LocalDateTime.parse("2022-01-03T00:00:00");
+                LocalDateTime dateNeeded = LocalDateTime.parse("2022-01-05T00:00:00");
+
+                RecommendationRequest recommendationRequest = RecommendationRequest.builder()
+                                .requesterEmail("supman@gmail.com")
+                                .professorEmail("drbob@ucsb.edu")
+                                .explanation("idk")
+                                .dateRequested(dateRequested)
+                                .dateNeeded(dateNeeded)
+                                .done(false)
```

Comment:
Need better tests; fail mutation testing (do not rely on github actions; run mutation testing on local machine!)